packages/browser/src/experimentClient.ts

```diff
@@ -107,27 +107,39 @@ export class ExperimentClient implements Client {
   /**
    * Returns the variant for the provided key.
    *
-   * Fetches {@link all} variants, falling back  on the given fallback, then the
-   * configured fallbackVariant.
+   * Access the variant from {@link Source}, falling back  on the given
+   * fallback, then the configured fallbackVariant.
+   *
+   * If an {@link ExperimentAnalyticsProvider} is configured, this function will
+   * call the provider with an {@link ExposureEvent}. The exposure event does
+   * not count towards your event volume within Amplitude.
    *
    * @param key The key to get the variant for.
    * @param fallback The highest priority fallback.
    * @see ExperimentConfig
+   * @see ExperimentAnalyticsProvider
    */
   public variant(key: string, fallback?: string | Variant): Variant {
     if (!this.apiKey) {
       return { value: undefined };
     }
+    const sourceVariant = this.sourceVariants()[key];
+    if (sourceVariant?.value) {
```

Comment:
Why not `if (sourceVariant) {`?

I had to check but we do currently require a value so this existing code is correct.

Reply:
Technically, sourceVariant could be non-null with a null value, although this is definitely not the regular case and I'd doubt it'd ever happen. That being said, a null value means an invalid variant, which we do not want to track.